src/modules/csm/distribution.py

```diff
@@ -0,0 +1,222 @@
+
+        log.distributable = rewards_to_distribute
+        log.distributed_rewards = sum(rewards_distribution.values())
+        log.rebate_to_protocol = rewards_to_distribute - log.distributed_rewards
+
+        return rewards_distribution, log
+
+    @lru_cache()
+    def _get_curve_params(self, curve_id: int, blockstamp: ReferenceBlockStamp):
+        perf_coeffs = self.w3.csm.params.get_performance_coefficients(curve_id, blockstamp.block_hash)
+        perf_leeway_data = self.w3.csm.params.get_performance_leeway_data(curve_id, blockstamp.block_hash)
+        reward_share_data = self.w3.csm.params.get_reward_share_data(curve_id, blockstamp.block_hash)
+        return perf_coeffs, perf_leeway_data, reward_share_data
+
+    def _get_network_performance(self, frame: Frame) -> float:
+        att_perf = self.state.get_att_network_aggr(frame)
+        prop_perf = self.state.get_prop_network_aggr(frame)
+        sync_perf = self.state.get_sync_network_aggr(frame)
+        network_perf = PerformanceCoefficients().calc_performance(att_perf, prop_perf, sync_perf)
+        return network_perf
+
+    @staticmethod
+    def process_validator_duties(
+        validator: LidoValidator,
+        attestation: DutyAccumulator | None,
+        sync: DutyAccumulator | None,
+        proposal: DutyAccumulator | None,
+        threshold: float,
+        reward_share: float,
+        perf_coeffs: PerformanceCoefficients,
+        participation_shares: defaultdict[NodeOperatorId, int],
+        log_operator: OperatorFrameSummary,
+    ) -> int:
+        if attestation is None:
+            # It's possible that the validator is not assigned to any duty, hence it's performance
+            # is not presented in the aggregates (e.g. exited, pending for activation etc).
+            # TODO: check `sync_aggr` to strike (in case of bad sync performance) after validator exit
+            return 0
+
+        log_validator = log_operator.validators[validator.index]
+
+        log_validator.threshold = threshold
+        log_validator.rewards_share = reward_share
+
+        if validator.validator.slashed is True:
+            # It means that validator was active during the frame and got slashed and didn't meet the exit
+            # epoch, so we should not count such validator for operator's share.
+            log_validator.slashed = True
+            return 0
+
+        performance = perf_coeffs.calc_performance(attestation, proposal, sync)
+
+        log_validator.performance = performance
+        log_validator.attestation_duty = attestation
+        if proposal:
+            log_validator.proposal_duty = proposal
+        if sync:
+            log_validator.sync_duty = sync
+
+        if performance > threshold:
+            # Count of assigned attestations used as a metrics of time the validator was active in the current frame.
+            # Reward share is a share of the operator's reward the validator should get. It can be less than 1.
```

Comment:
An example needed, e.g., given only 1 attestation made by a validator and `rewards_share < 1` we would end up sending everything to the treasury as a rebate. We cover this case in favor of operator.

If I get it right. If not, it's another reason to have an explainer here.

Reply:
There should be `math.ceil`. I will fix it and add explainer

Reply:
Done